index.bs

```diff
@@ -790,7 +791,12 @@ To <dfn>initialize WebTransport over HTTP</dfn>, given a {{WebTransport}} object
   1. Let |connection| be the result of [=obtain a connection|obtaining a connection=] with
      |networkPartitionKey|, |url|, false, |newConnection|, and |http3Only|. If the user agent
      supports more than one congestion control algorithm, choose one appropriate for
-     |congestionControl| for sending of data on this |connection|.
+     |congestionControl| for sending of data on this |connection|. When obtaining a connection, if
+     |serverCertificateHashes| is specified, instead of using the default certificate verification
+     algorithm, consider the certificate valid if it meets the [=custom certificate
+     requirements=] and if [=verify a certificate hash|verifying the certificate hash=] against
+     |serverCertificateHashes| returns true. If either condition is not met, let |connection| be
+     failure.
   1. If |connection| is failure, then abort the remaining steps and [=queue a network task=] with
```

Comment:
The phrasing here is a bit odd.  Maybe `if |connection| failed for any reason`.

Reply:
"[Obtain a connection](https://fetch.spec.whatwg.org/#concept-connection-obtain)" actually returns "failure" if the connection fails, which is what this step is checking.

Reply:
Ah, I see what you are doing.  This is like a Rust `Result<T, E>` where you get a composite object that is either success (a connection) or an error.  You are checking if the result was an error.

The problem I have is that - while this fine if you are writing code, this is terribly stilted prose.  It reads as a grammatical error.

Reply:
Yes, while I agree that it sounds odd, I think this is a fairly standard pattern throughout the spec ecosystem. For example, step 2 of https://url.spec.whatwg.org/#concept-url-parser and step 2 of https://encoding.spec.whatwg.org/#dom-textdecoder have similar text saying "if _x_ is failure". There's also an open issue about clarifying what "failure" is at https://github.com/whatwg/infra/issues/545, so perhaps when that is fixed, we can update it in the WebTransport spec.